implementations/rust/ockam/ockam_api/src/cloud/share/accept.rs

```diff
@@ -0,0 +1,85 @@
+use minicbor::{Decode, Encode};
+use serde::Serialize;
+
+use super::*;
+
+#[derive(Clone, Debug, Decode, Encode, Serialize)]
+#[cbor(map)]
+#[rustfmt::skip]
+pub struct AcceptInvite {
+    #[n(1)] pub id: String,
+}
+
+#[derive(Clone, Debug, Decode, Encode, Serialize)]
+#[cbor(map)]
+#[rustfmt::skip]
+pub struct AcceptedInvite {
+    #[n(1)] pub id: String,
+    #[n(2)] pub scope: RoleInShare,
+    #[n(3)] pub target_id: String,
+}
+
+mod node {
+    use ockam_core::api::{Request, Response};
+    use ockam_core::{self, Result};
+    use ockam_multiaddr::MultiAddr;
+    use ockam_node::Context;
+
+    use crate::cloud::CloudRequestWrapper;
+    use crate::nodes::{NodeManager, NodeManagerWorker};
+
+    use super::*;
+
+    const API_SERVICE: &str = "users";
+
+    impl NodeManager {
+        pub async fn accept_invite(
+            &self,
+            ctx: &Context,
+            req: AcceptInvite,
+            route: &MultiAddr,
+            identity_name: Option<String>,
+        ) -> Result<AcceptedInvite> {
+            Response::parse_response_body(
+                self.accept_invite_response(ctx, CloudRequestWrapper::new(req, route, identity_name))
+                    .await?
+                    .as_slice(),
+            )
+        }
+
+        pub(crate) async fn accept_invite_response(
+            &self,
+            ctx: &Context,
+            req_wrapper: CloudRequestWrapper<AcceptInvite>,
+        ) -> Result<Vec<u8>> {
+            let cloud_multiaddr = req_wrapper.multiaddr()?;
+            let req_body = req_wrapper.req;
+            let label = "accept_share";
+            let req_builder = Request::post("/v0/redeem_invite").body(req_body);
+
+            self.request_controller(
+                ctx,
+                label,
+                None,
+                &cloud_multiaddr,
+                API_SERVICE,
+                req_builder,
+                // TODO: Should this include an identifier?
+                None,
```

Comment:
If you don't pass an identifier the default identity that is associated to the `NodeManager` will be used to start a secure channel with the `Orchestrator`